entity-service/src/main/kotlin/com/egm/stellio/entity/config/Neo4jConfiguration.kt

```diff
@@ -1,21 +1,24 @@
 package com.egm.stellio.entity.config
 
-import org.neo4j.ogm.session.SessionFactory
+import org.springframework.boot.autoconfigure.data.neo4j.Neo4jProperties
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Profile
 import org.springframework.data.neo4j.repository.config.EnableNeo4jRepositories
-import org.springframework.data.neo4j.transaction.Neo4jTransactionManager
-import org.springframework.transaction.annotation.EnableTransactionManagement
 
+@Profile("!test")
 @Configuration
 @EnableNeo4jRepositories(basePackages = ["com.egm.stellio.entity.repository"])
-@EnableTransactionManagement
-class Neo4jConfiguration(
-    private val sessionFactory: SessionFactory
-) {
+class Neo4jConfiguration {
 
     @Bean
-    fun transactionManager(): Neo4jTransactionManager {
```

Comment:
what is the rationale behind removing the transaction manager (and management)?

Reply:
similar to [this](https://github.com/neo4j/neo4j-java-driver-spring-boot-starter/issues/10) issue, we just need to configure the OGM starter.
Otherwise, it would be like a double configuration and there is a cyclic dependency between the manual OGM configuration and the sessionFactory.
I tried to keep the Neo4jTransactionManager but the application failed to start due to the mentionned dependencies conflict.

Reply:
well, not exactly the same, because we were not creating a new session factory (but using the existing one to create the transaction manager, based on what is said in [SDN manual](https://docs.spring.io/spring-data/data-neo4j/docs/current/reference/html/#reference.getting_started.spring-configuration): "Note that you will have to activate @EnableTransactionManagement explicitly to get annotation based configuration at facades working as well as define an instance of this Neo4jTransactionManager with the bean name transactionManager".

but if you can guarantee that transactions work as expected at the service level, I'll buy it.